### PR TITLE
Final changes

### LINKED ISSUE #165: UG: adding events - specify to user
specify that the start date follows that in the original command instead of what they type in after

### PR DESCRIPTION
**Pull request by NicholasTanYY**
Resolves #165 #171 #172 #191 #192

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -387,6 +387,18 @@ ics, import, tasks.ics
...
+**A**: Not to worry! Our program registers the start date as the date you keyed into the original command, i.e. the first day of the month. The subsequent prompt for the start date is for formatting purposes and will not affect the actual date of the event.
+
+**Q**: I have an event lasting from 15 Apr to 19 Apr within a week. Why is it that `free, 16` shows that the day is free?
+
+**A**: For events which stretch over multiple days, we only consider the original day of the event when calculating free times. In this case, the event starts on the 15th and ends on the 19th, so the 16th, 17th, 18th and 19th are considered free days. Our advise: add events with their corresponding timings for each day of the event to get a more accurate representation of your schedule.
+
+**Q**: Does the scheduler account for clashes in event timings?
+
+**A**: No. This is to accomodate for users who may have overlapping events or tasks. We recommend users to manually check for clashes and adjust their schedules accordingly.
+
```
Comment (clarencepohh) [inline docs/UserGuide.md:401]: Nice Q&A section!

File: docs/UserGuide.md [hunk 2]
```diff
@@ -398,18 +410,19 @@ For a quick reference, here's a summary of available commands:
...
-- `Event`: A task that has a start and end date / time.
+- Todo (`T`): A basic task with no start or end date / times.
+- Deadline (`D`): A task that has a date / time to complete by.
+- Event (`E`): A task that has a start and end date / time.
 
 Tasks can be classified in the following ways:
-- marked as `HIGH`, `MEDIUM` or `LOW` priority. Default level is `LOW`.
-- marked as `COMPLETE` or `INCOMPLETE`. Default is `INCOMPLETE`.
+- marked as HIGH (`H`), MEDIUM (`M`) or LOW (`L`) priority. Default level is LOW.
+- marked as COMPLETE (`X`) or INCOMPLETE (`O`). Default is INCOMPLETE.
```
Comment (clarencepohh) [inline docs/UserGuide.md:426]: Thanks for removing the code blocks that may cause confusion

File: docs/team/nicholastanyy.md
Comment (clarencepohh) [inline docs/team/nicholastanyy.md:1]: Your contributions to the team really shine through in your PPP 😮 Thanks for being on the team!

File: docs/images/sequence/PrintWeekViewSequenceDiagram.png
Comment (clarencepohh) [inline docs/images/sequence/PrintWeekViewSequenceDiagram.png:1]: Nice job fixing the errors pointed out by our TA!

**Review by clarencepohh — APPROVED** [review]
Good fixes overall. Thanks!

File: docs/diagrams/class/UiRendererClassDiagram.puml [hunk 3]
```diff
@@ -0,0 +1,54 @@
+@startuml
+
```
Comment (Mahesh1772) [inline docs/diagrams/class/UiRendererClassDiagram.puml:2]: Thanks for creating a UML class diagram!

File: docs/images/class/Ui.png
Comment (Mahesh1772) [inline docs/images/class/Ui.png:1]: Good neat diagram!

File: docs/team/nicholastanyy.md [hunk 4]
```diff
@@ -1,12 +1,56 @@
...
+<br><br>
+These were done by having a class which calculates the dimensions of the terminal. This code is not included in the unused folder as it is a code snippet modified from something I found online. The code uses an intelligent way of finding terminal dimensions, using ANSI codes to control the position of the cursor. Then using a TerminalController class (in unused code segment), I was able to send keyboard inputs to the terminal to fix some issues with the outputs to the terminal that arose during the calculation of terminal dimensions.
+<br><br>
+I successfully managed to centralise some components and fit corresponding avatar sizes but ran into major issues when trying to run the text-ui-tests. This feature took a lot of effort to implement. I had to abandon this feature as I was worried that it could cause more problems to other implementations in the future, like not working in other terminal types or conflicting with other Ui implementations.
+
+## Reflections
+This project has been a great experience and a first step into the software engineering world for my teammates and I. I look forward to learning from this experience and build more software in the future!
```
Comment (Mahesh1772) [inline docs/team/nicholastanyy.md:56]: Thanks for being a good teammate!

File: text-ui-test/input.txt [hunk 5]
```diff
@@ -59,4 +59,13 @@ add, 17, D, new deadline 2
 priority,17,2,M
 update, 17, 1, qioncioqnc uion uioqen qwpc
 delete, 17, 4
+week
```
Comment (Mahesh1772) [inline text-ui-test/input.txt:62]: Good job on adding extensive testing in the input text

File: docs/UserGuide.md [hunk 6]
```diff
@@ -387,6 +387,18 @@ ics, import, tasks.ics
 
 **A**: You can transfer your data to another computer by copying the data file saved in the CLI-nton application directory. The data file is typically named `clintonData.txt` and contains all your tasks. Simply copy this file to the same location on the other computer to transfer your tasks.
 
+**Q**: When adding an Event task, I entered `add,1,E,Event 1` to add an event on the first day of the month, but after that, I was prompted to enter the start date again. Why is that?
+
+**A**: Not to worry! Our program registers the start date as the date you keyed into the original command, i.e. the first day of the month. The subsequent prompt for the start date is for formatting purposes and will not affect the actual date of the event.
+
+**Q**: I have an event lasting from 15 Apr to 19 Apr within a week. Why is it that `free, 16` shows that the day is free?
```
Comment (Mahesh1772) [inline docs/UserGuide.md:394]: Nice job on adding more Q and A to clarify any lingering doubts!

**Review by Mahesh1772 — COMMENTED** [review]
LGTM!

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -67,16 +68,15 @@ The UiRenderer component is responsible for rendering the user interface. It is
...
-### printWeekView Method
-![Sequence Diagram for printWeekView()](images/sequence/PrintWeekViewSequenceDiagram.png)
+### printView Method for Week View
+![Sequence Diagram for printView()](images/sequence/PrintWeekViewSequenceDiagram.png)
 
 ### printWeekHeader Method
 `printWeekHeader` Method
 
-The `printWeekHeader` method is responsible for rendering the header section of the week view, including the names of the days and optionally the dates.
+The `printWeekHeader` method is responsible for rendering the header section of the week and month view, including the names of the days and the dates.
```
Comment (kyhjonathan) [inline docs/DeveloperGuide.md:79]: Thanks for changing the wordings to make it more explicit!

File: docs/diagrams/class/UiRendererClassDiagram.puml [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+skinparam classAttributeIconSize 0
```
Comment (kyhjonathan) [inline docs/diagrams/class/UiRendererClassDiagram.puml:3]: UML class diagram LGTM!

File: src/test/java/ui/AvatarUiTest.java [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+class AvatarUiTest {
```
Comment (kyhjonathan) [inline src/test/java/ui/AvatarUiTest.java:12]: Nice JUnit tests for AvatarUI!

File: src/test/java/ui/UiRendererTest.java [hunk 10]
```diff
@@ -40,4 +65,109 @@ void printTaskForDay_noTasks_printsNoTasks() {
...
+    void printWeekHeader_forWeekView_printsWeekViewHeading() {
+        LocalDate today = LocalDate.now();
+        LocalDate startOfWeek = DateUtils.getStartOfWeek(today);
+        DateTimeFormatter dateFormatter = DateTimeFormatter.ofPattern("dd/MM/yyyy");
+        ByteArrayOutputStream outContent = new ByteArrayOutputStream();
+        System.setOut(new PrintStream(outContent));
+
+        printWeekHeader(startOfWeek, dateFormatter, false);
+
+        String expectedOutput =
```
Comment (kyhjonathan) [inline src/test/java/ui/UiRendererTest.java:78]: Will the expected Output always be in this format? 
e.g. If the user does testing in 2025, will the expected output be the text given below?

File: src/test/java/ui/UiRendererTest.java [hunk 11]
```diff
@@ -40,4 +65,109 @@ void printTaskForDay_noTasks_printsNoTasks() {
...
+                "|Task 1         |Task 2         |               |               |" + 
+                "               |               |               |\n" +
+                "+---------------+---------------+---------------+---------------+" + 
+                "---------------+---------------+---------------+\n";
+
+        assertEquals(expectedOutput, outContent.toString());
+    }
+
+    @Test
+    void printWeekBody_multipleTasksPerDay_printsWeekBodyWithMultipleTasksPerDay() throws TaskManagerException {
```
Comment (kyhjonathan) [inline src/test/java/ui/UiRendererTest.java:147]: LGTM!

**Pull request closed without merging**